source/distributed/shared_tria.cc

```diff
@@ -319,6 +323,50 @@ namespace parallel
 
 
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::set_subdomain_mask(
+      const SubdomainMask new_mask)
+    {
+      if (allow_artificial_cells && new_mask != subdomain_mask)
```

Comment:
I am not sure here. Do we need `allow_artificial_cells` any longer? I think we might be able to switch dynamically with the new function?